pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -613,6 +619,72 @@ func (r *ReconcileComplianceRemediation) verifyAndCompleteMC(obj *unstructured.U
 	return nil
 }
 
+// Process kubeletconfig remediation
+func (r *ReconcileComplianceRemediation) verifyAndCompleteKC(obj *unstructured.Unstructured, rem *compv1alpha1.ComplianceRemediation) error {
+	scan := &compv1alpha1.ComplianceScan{}
+	scanKey := types.NamespacedName{Name: rem.Labels[compv1alpha1.ComplianceScanLabel], Namespace: rem.Namespace}
+	if err := r.client.Get(context.TODO(), scanKey, scan); err != nil {
+		return fmt.Errorf("couldn't get scan for KC remediation: %w", err)
+	}
+	mcfgpools := &mcfgv1.MachineConfigPoolList{}
+	if err := r.client.List(context.TODO(), mcfgpools); err != nil {
+		return fmt.Errorf("couldn't list the pools for the remediation: %w", err)
+	}
+	// The scans contain a nodeSelector that ultimately must match a machineConfigPool. The only way we can
+	// ensure it does is by checking if it matches any MachineConfigPool's labels.
+	// See also: https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md
+	ok, pool := utils.AnyMcfgPoolLabelMatches(scan.Spec.NodeSelector, mcfgpools)
+	if !ok {
+		return common.NewNonRetriableCtrlError("not applying remediation that doesn't have a matching MachineconfigPool. Scan: %s", scan.Name)
+	}
+
+	// We need to check if there is exsisting kubeletConfigMC present for selected MCP
+	hasCustomKC, kubletMCName, err := utils.IsMcfgPoolUsingKC(pool)
+
+	if err != nil {
+		//string to int convertion error
+		return fmt.Errorf("string-int convertion error for KC remediation: %w", err)
```

Comment:
General golang nit: the fact that the error is a str2int should be expressed in err. Ideally, you would just return the error, wrapping it in context that this level knows about. This is also future-proof in case the error would mean something else in the future.

Reply:
That's a very good point.